src/modal/modal.js

```diff
@@ -463,6 +522,25 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap'])
         appendToElement.append(modalDomEl);
         appendToElement.addClass(modalBodyClass);
 
+        // if modal is attached to body, then check if body is overflowing and set bodyOverflowing property on modal object and set right padding equal to scrollbarWidth
+        if (modal.appendTo === 'body' && document.body.scrollHeight > document.documentElement.clientHeight){
+
+          // set bodyOverflowing property
+          modalDomEl.bodyOverflowing = true;
+
+          // set right padding to body
+          if(openedWindows.length() === 1){
+            if(!$$scrollbarHelper.scrollbarWidth){
+              $$scrollbarHelper.measureScrollbar();
+            }
+
+            var body = $document.find('body');
+            // set originalBodyRightPadding property to reset it when the last modal closes
+            modalDomEl.originalBodyRightPadding = body.css('padding-right');
```

Comment:
That is an unnecessarily verbose key name - please choose a better name, such as `bodyRightPadding`.